bcda/auth/client/oktaclient.go

```diff
@@ -258,6 +254,49 @@ func refreshKeys() {
 	}
 }
 
+func (oc *OktaClient) GenerateNewClientSecret(clientID string) (string, error) {
+	url := os.Getenv("OKTA_CLIENT_ORGURL") + "/oauth2/v1/clients/" + clientID + "/lifecycle/newSecret"
+
+	reqID := uuid.NewRandom()
+	logRequest(reqID).WithFields(logrus.Fields{"url": url, "clientID": clientID}).Print()
+	req, err := http.NewRequest("POST", url, nil)
+	if err != nil {
+		logError(err, reqID)
+		return "", err
+	}
+
+	addRequestHeaders(req)
+
+	resp, err := client().Do(req)
+	if err != nil {
+		logError(err, reqID).Print()
+		return "", err
+	}
+	if resp.StatusCode >= 400 {
+		logError(errors.New(resp.Status), reqID).Print()
+		return "", errors.New(resp.Status)
+	}
+
+	logResponse(resp.StatusCode, reqID).Print()
+	var result map[string]interface{}
+	err = json.NewDecoder(resp.Body).Decode(&result)
+	if err != nil {
+		return "", err
+	}
+	cs := result["client_secret"].(string)
+
+	return cs, nil
+}
+
+func client() *http.Client {
+	return &http.Client{Timeout: time.Second * 10}
+}
+func addRequestHeaders(req *http.Request) {
+	req.Header.Set("Accept", "application/json")
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "SSWS "+os.Getenv("OKTA_CLIENT_TOKEN"))
```

Comment:
This one should use the package global `oktaAuthString` because it's set once in config.